### PR TITLE
Improved schema_generator tool

### LINKED ISSUE #11674: Improve `schema_generator.py` developer experience
## Tell us about the problem you're trying to solve
When building a connector, one of the most important requirements is to define the output schema of each stream in the connector. There are multiple ways of doing this: 
1. Transcribing the JSONSchema by hand based on reading API docs
2. Using the API's OpenAPI spec to discern the schema. [Docs](https://docs.airbyte.com/connector-development/cdk-python/schemas#generating-schemas-from-openapi-definitions)
3. reading data from the API and reverse engineering it using `./tools/integrations/schema_generator.py`

Solution 3, while it has its limitations, is convenient in many cases as it allows you to get a schema that should be mostly correct for your connector. However, it has many tactical drawbacks: 
1. It's not documented on the [CDK schemas page](https://docs.airbyte.com/connector-development/cdk-python/schemas) as a viable option
2. It doesn't leverage python virtual environments, which means the user must install packages (specifically, `genson`) in the global namespace to use the utility
3. There is no readme or doc describing how to use the tool (it works by piping the output of a connector into it e.g: `docker run <connector> read --config <path>  --catalog <path> | python ./tools/integrations/schema_generator.py`, but you need to read the code to make that guess)
4. It's a standalone tool that requires downloading the entire Airbyte monorepo to be used
5. It requires running python

## Describe the solution you’d like

Solving the points above: 

1. Package this as part of the Python CDK. I should be able to install only the CDK (`pip install airbyte_cdk`), then run a specific CDK command e.g: `cat records.txt | python -m airbyte_cdk infer_schemas`. Some things you should scope out as part of this ticket are: where in the CDK file structure should this go (probably in a utils/ package), and where the output schemas JSONs should go (currently they're just output to a new folder `schemas/` of the working directory from which you run that command)
2. if I use the tool incorrectly or ask for help e.g: `python schema_generator.py herp` or `python schema_generator.py --help` it should print a usage message describing how I'm supposed to use it
4. (optional, maybe just create a follow-up ticket for now) it should be runnable via docker image so non-python programmers find it accessible
5. Once the points above are solved, document it in the CDK schemas documentation and the CDK README

This ticket should definitely be split into separate subtasks/PRs each addressing one or at most two of the points above at a time. 

## Some problems to consider as you solve this problem
`genson` (the core package in this script) is only compatible (it seems?) with Python <=3.8 but the CDK uses 3.9. The 20/80 version of this ticket may be to: 
1. package this up in its own virtual env, backed with a setup.py etc.. -- it would still live in `tools/` but it would be in its own dedicated directory + virtualenv to run it (which is a decent place to be)
2. document the usage of this in both the docs & the code runner itself (like mentioned above, add a `USAGE` message)

### PR DESCRIPTION
**Pull request by dougpm**
## What
Improves the developer experience when using the schema generator tool.

## How
1. Created a package for the tool.
2. Added a command for generating a Configured Catalog.
3. Added documentation on how the tool works and how to use it.

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: tools/schema_generator/schema_generator/infer_schemas.py [hunk 1]
```diff
@@ -1,27 +1,3 @@
-#
-# MIT License
```
Comment (koconder) [inline tools/schema_generator/schema_generator/infer_schemas.py:2]: I dont think you can drop the licence

File: tools/schema_generator/schema_generator/main.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#
+
+import argparse
+import sys
+
+from .infer_schemas import infer_schemas
+from .configure_catalog import configure_catalog
+
+
+def main():
```
Comment (koconder) [inline tools/schema_generator/schema_generator/main.py:12]: Any unit tests for the tool?

**Review by vincentkoc — CHANGES_REQUESTED** [review]
Good stuff, i think you need to add unit tests and re-instate the licence. I only did a quick eye-ball on the logic

**Comment by alafanechere** [comment]
Thank you @dougpm for this great contribution! Our team will review it soon.

**Comment by alafanechere** [comment]
Hi @dougpm let me know if I'm wrong:
* Your tool is generating schemas files from running a `read` command
* The `read` command requires a catalog to run, hence the underlying stream schema

So I'm under the impression that this tool can't be used to initially create schema files while starting a new connector project. Am I right?

**Comment by dougpm** [comment]
Hey @alafanechere 

You're right, to generate the catalog we need the schema files. These files can be empty json files, they just need to be there for the `discover` and `read` commands to succeed.
I could add instructions for the creation of empty schema files for each stream before the connector container is built. 
Should we go with that or are there any other options?

**Comment by alafanechere** [comment]
Hey @dougpm I pinged our Developer Experience team to get their opinion about this addition 😄

**Comment by alafanechere** [comment]
@dougpm it would help if you can add a bit more details in your documentation about the context in which this tool can be used in the connector development cycle.

**Comment by dougpm** [comment]
@alafanechere Added a __Prerequisites__ section in the tool README file.

**Review by sherifnada — APPROVED** [review]
Generally looks good, thanks for adding docs! would also support having unit tests here but otherwise LGTM

**Review by alafanechere — CHANGES_REQUESTED** [review]
@dougpm how do you feel about adding unit tests?

**Comment by alafanechere** [comment]
Hey @dougpm do you have the bandwidth to implement the unit tests soon?

**Comment by dougpm** [comment]
Hey @alafanechere yes, I'll do it this week, sorry for the delay.

**Comment by dougpm** [comment]
@alafanechere tests added!

**Review by alafanechere — APPROVED** [review]
Thank you for adding the tests. I played with the schema generator locally, I had to make minor changes to the installation process to make it work.

**Pull request merged**